infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -135,23 +130,23 @@ public LogUnitServer(ServerContext serverContext) {
      */
     public LogUnitServer(ServerContext serverContext, LogUnitServerInitializer serverInitializer) {
         this.serverContext = serverContext;
-        config = LogUnitServerConfig.parse(serverContext.getServerConfig());
-        executor = serverContext.getExecutorService(serverContext.getLogUnitThreadCount(), "LogUnit-");
+        executor = serverContext.getExecutorService(serverContext.getConfiguration().getNumLogUnitWorkerThreads(),
```

Comment:
when you pass an object which has side effects and/or communicates to the external world or keeps some state, it affects your code and makes it harder to write tests to keep the code flexible and loosely coupled. 

That was the motivation for passing a config instead of a bigger and quite heavy serverContext object.

Next time when you'll need to write a unit test for LogUnitServer, it will be more complicated to do that if you'd use serverContext.

Reply:
So, would be great to have only necessary dependencies in the classes

Reply:
got it, so capture snapshot of necessary attributes as the config object in the tests, and pass that state around?

Reply:
In this case, probably you'll need to create an additional constructor, which seems outside of the scope.
Sorry, if there any confusion from my previous comments :) I thought LogUnitServer used a config in its constructor

Reply:
For unit tests, as those in `infrastructure/LogUnitServerTest`, we already decoupled it using mocks, so might not be a problem for adding more unit tests. But yeah I agree that passing ServerContext is too heavy and might have potential problems.